R/bundle.R

```diff
@@ -406,14 +406,32 @@ inferAppMode <- function(appDir, appPrimaryDoc, files, quartoInfo) {
     return("shiny")
   }
 
-  # Determine if we have Rmd and if they are (optionally) need the Shiny runtime.
-  rmdFiles <- grep("^[^/\\\\]+\\.[rq]md$", files, ignore.case = TRUE, perl = TRUE, value = TRUE)
+  # Determine if we have Rmd files, and if they use the Shiny runtime.
+  rmdFiles <- grep("^[^/\\\\]+\\.rmd$", files, ignore.case = TRUE, perl = TRUE, value = TRUE)
   shinyRmdFiles <- sapply(file.path(appDir, rmdFiles), isShinyRmd)
 
-  # An Rmd file with a Shiny runtime uses rmarkdown::run.
+  # Determine if we have qmd files, and if they use the Shiny runtime
+  qmdFiles <- grep("^[^/\\\\]+\\.qmd$", files, ignore.case = TRUE, perl = TRUE, value = TRUE)
+  shinyQmdFiles <- sapply(file.path(appDir, qmdFiles), isShinyRmd)
+
+  # Trying to deploy Quarto and R Markdown files simultaneously is an error.
+  if (length(rmdFiles) > 0 && length(qmdFiles) > 0) {
```

Comment:
We should think about this error a bit more. It is possible to have a Quarto project that uses a `document.Rmd` file. Similarly, it is possible to have a Quarto project with a `notebook.ipynb` file.

It might be helpful if we captured an enumeration of file name collections and explained which combinations lead to different app modes, especially in the presence / non-presence of `quartoInfo`.

Here's one example:

* `_quarto.yml`
* `analysis.Rmd`
* `playground.ipynb`

What content type should be used when this code is deployed to Connect? Why?

Similarly, what about:

* `report.Rmd` with non-nil `quartoInfo`

Is this information more a signal to use Quarto or to use R Markdown?

Reply:
Yeah, you raise a good point.

The app mode is inferred _after_ we have put Quarto info into the format we desire, so by this point it's not clear whether it comes from running `quarto inspect` or as metadata passed in from the Quarto package.

I'm not sure what Connect would do with the first example you provide.

We had been using `quartoInfo`'s presence as an indicator of user intent to deploy Quarto. If continue to rely on that signal, we could do something like the following.

- with `quartoInfo`:
    - App mode is `quarto-static`, or `quarto-shiny` if any `shinyRmdFiles` or `shinyQmdFiles`.
- without `quartoInfo`:
    - App mode is `rmd-static` or `rmd-shiny`, as would have been otherwise detected
    - Any `qmdFiles` or `shinyQmdFiles` results in an error?

(Thinking about this more. When Quarto renders R Markdown files, their chunk arguments are parsed as R Markdown would, I'm guessing.)